request-server_test.go

```diff
@@ -354,7 +351,9 @@ func TestRequestReaddir(t *testing.T) {
 	for i := 0; i < 100; i++ {
 		fname := fmt.Sprintf("/foo_%02d", i)
 		_, err := putTestFile(p.cli, fname, fname)
-		assert.Nil(t, err)
+		if !assert.NoError(t, err) {
```

Comment:
Since you‘re calling `t.Fatal` below which will log the error anyways, this would probably be clearer as:
```
if err != nil {
  t.Fatal("expected no error, got:", err)
}
```

Reply:
Another good suggestion.